### PR TITLE
Additional tests for third-party event rules

### PR DESCRIPTION
**Pull request by richvdh**
~~Based on #8467~~

### CODE REVIEW

File: tests/rest/client/test_third_party_rules.py [hunk 1]
```diff
@@ -12,49 +12,57 @@
...
+thread_local = threading.local()
+
 
 class ThirdPartyRulesTestModule:
-    def __init__(self, config, *args, **kwargs):
-        pass
+    def __init__(self, config, module_api):
+        # keep a record of the "current" rules module, so that the test can patch
+        # it if desired.
+        thread_local.rules_module = self
```
Comment (anoadragon453) [inline tests/rest/client/test_third_party_rules.py:33]: Do you use `thread_local` here as we could be running the tests in parallel?

File: tests/rest/client/test_third_party_rules.py [hunk 2]
```diff
@@ -76,6 +92,16 @@ def test_third_party_rules(self):
         self.render(request)
         self.assertEquals(channel.result["code"], b"200", channel.result)
 
+        callback.assert_called_once()
+
+        # there should be various state events in the state arg: do some basic cheks
```
Comment (anoadragon453) [inline tests/rest/client/test_third_party_rules.py:97]:
```suggestion
        # there should be various state events in the state arg: do some basic checks
```

**Review by anoadragon453 — APPROVED** [review]

**Comment by anoadragon453** [comment]
Oh, though I've failed to notice the CI failures.

File: changelog.d/8468.misc [hunk 3]
```diff
@@ -0,0 +1 @@
+Additional testing for ThirdPartyEventRules.
```
Comment (anoadragon453) [inline changelog.d/8468.misc:1]:
```suggestion
Additional testing for `ThirdPartyEventRules`.
```

File: tests/rest/client/test_third_party_rules.py [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+        )
+        self.render(request)
+        self.assertEquals(channel.result["code"], b"403", channel.result)
+
+    def test_modify_event(self):
+        """Tests that the module can successfully tweak an event before it is persisted.
+        """
+        # first patch the event checker so that it will modify the event
+        async def check(ev: EventBase, state):
+            ev.content = {"x": "y"}
```
Comment (anoadragon453) [inline tests/rest/client/test_third_party_rules.py:119]: This just needs to return `True` in addition to modifying the event.

**Review by anoadragon453 — CHANGES_REQUESTED** [review]
Coffee acquired.

File: tests/rest/client/test_third_party_rules.py (hunk 1 above)
Comment (richvdh) [reply tests/rest/client/test_third_party_rules.py:33]: yeah, that was the general thought.

**Review by anoadragon453 — APPROVED** [review]

**Pull request merged**